src/main/java/org/terracotta/statistics/derived/OperationResultSampler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * All content copyright Terracotta, Inc., unless otherwise indicated.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terracotta.statistics.derived;
+
+import org.terracotta.statistics.observer.ChainedEventObserver;
+
+import java.util.Set;
+import java.util.concurrent.ThreadLocalRandom;
+
+/**
+ * @author cdennis
+ */
+public class OperationResultSampler<T extends Enum<T>> extends OperationResultFilter<T> {
+
+  private final int ceiling;
+
+  public OperationResultSampler(Set<T> targets, double sampling, ChainedEventObserver... observers) {
+    super(targets, observers);
+    if (sampling > 1.0 || sampling < 0.0) {
+      throw new IllegalArgumentException();
```

Comment:
Some message in the exception please

Reply:
This class has just been renamed from @chrisdennis's code : I'll add a message though ;-)